### ISSUE TITLE
Typed function containing only a docstring does not have any error raised about its return type not being met

### ISSUE THREAD

**Issue by brettcannon** [microsoft/pylance-release#2111]
## Environment data

-   Language Server version: v2021.11.3-pre.1
-   OS and version: Windows 11
-   Python version (& distribution if applicable, e.g. Anaconda): CPython 3.10.0

## Expected behaviour

Example function:
```python
def entity_to_comment(entity: str, properties: set[Property]) -> str:
    """Convert an entity and its properties to a `__GDPR__` comment.""
```

I would expect to receive some error message about no return value for a function with a defined return type whenever the return type is not `None`.

## Actual behaviour

No error message. You can add a `pass` line to see the expected error message.

## Logs

```
XXX
```

## Code Snippet / Additional information

Problematic code:
```python
def entity_to_comment(entity: str, properties: set[Property]) -> str:
    """Convert an entity and its properties to a `__GDPR__` comment."""
```

Code that shows the expected error (notice the added `pass` line to trigger the error message):
```python
def entity_to_comment(entity: str, properties: set[Property]) -> str:
    """Convert an entity and its properties to a `__GDPR__` comment."""
    pass
```

**Comment by erictraut** [comment]
We currently exempt functions and methods from return type consistency checks if there are no statements present in the function's body other than `...` or doc strings. The reasoning is that this is a common pattern for abstract methods and methods in protocols.

I can see why this would be unexpected behavior though. Perhaps we should do one or more of the following:
1. Apply the exemption only to methods, not to functions
2. Apply the exemption only to methods within a class that inherits from Protocol or from ABC
3. Apply the exemption only if there's an ellipses statement present (i.e. a doc string by itself is not sufficient)

I'm thinking that perhaps 1 _and_ 3 would be the best approach. Thoughts?

**Comment by erictraut** [comment]
Thinking about this a bit more, I think we should exempt any function (even non-methods) that are implemented with a `...` either with or without a docstring. A docstring by itself should not be sufficient in any case to exempt the function/method from a return type consistency check.

This change will be in the next release.

**Comment by brettcannon** [comment]
@erictraut thanks!

**Comment by debonte** [comment]
This issue was fixed in version 2021.12.0. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2021120-2-December-2021)
